src/traces/scattergeo/hover.js

```diff
@@ -0,0 +1,105 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Fx = require('../../plots/cartesian/graph_interact');
+var Axes = require('../../plots/cartesian/axes');
+
+var getTraceColor = require('../scatter/get_trace_color');
+var attributes = require('./attributes');
+
+
+module.exports = function hoverPoints(pointData) {
+    var cd = pointData.cd,
+        trace = cd[0].trace,
+        xa = pointData.xa,
+        ya = pointData.ya,
+        geo = pointData.subplot;
+
+    if(cd[0].placeholder) return;
+
+    function c2p(lonlat) {
+        return geo.projection(lonlat);
+    }
+
+    function distFn(d) {
+        var lonlat = d.lonlat;
+
+        // this handles the not-found location feature case
+        if(lonlat[0] === null || lonlat[1] === null) return Infinity;
+
+        if(geo.isLonLatOverEdges(lonlat)) return Infinity;
+
+        var pos = c2p(lonlat);
+
+        var xPx = xa.c2p(),
+            yPx = ya.c2p();
+
+        var dx = Math.abs(xPx - pos[0]),
+            dy = Math.abs(yPx - pos[1]),
+            rad = Math.max(3, d.mrc || 0);
```

Comment:
Do you think it will be obvious to someone who is maintaining this code what the `3` and `0` are doing? I didn't get it right away which is usually a sign that a comment may be helpful